File: 3838-path-existence-queries-in-a-graph-i/3838-path-existence-queries-in-a-graph-i.py
class Solution:
    def pathExistenceQueries(self, n: int, nums: List[int], maxDiff: int, queries: List[List[int]]) -> List[bool]:

        #Freidy the frog (SORTED hence..)

        cluster_list = [0] * n
        cluster = 0
        res = []
        for i in range(1, n):
            if abs(nums[i]- nums[i-1]) > maxDiff:
                cluster +=1
            cluster_list[i] = cluster

        for u,v in queries:
            res.append(cluster_list[u] == cluster_list[v])
        return res

File: 3838-path-existence-queries-in-a-graph-i/3838-path-existence-queries-in-a-graph-i.cpp
class Solution {
public:
       vector<bool> pathExistenceQueries(int n, vector<int>& A, int maxDiff, vector<vector<int>>& queries) {
        vector<int> g(n, 0);
        for (int i = 1; i < n; ++i) {
            g[i] = g[i - 1] + ((A[i] - A[i - 1]) > maxDiff ? 1 : 0);
        }
        
        vector<bool> res;
        for (const auto& q : queries) {
            int i = q[0], j = q[1];
            res.push_back(g[i] == g[j]);
        }
        return res;
    }
};